### PR TITLE
[ioctl] Build jwtsign command line into new ioctl

### LINKED ISSUE #3287: [ioctl] build jwtsign command line into new ioctl
### What would you like to be added
 - Step 1: create `jwtsign.go` and `jwtsign_test.go` files in `./ioctl/newcmd/jwt`. create this directory if not exist. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/jwt/jwtsign.go in `jwtsign.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/jwt/jwtsign.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description
Refactor `jwtsign` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3287 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewJwtSignCmd

**Test Configuration**:
- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
@huof6829 The unittest need to add signer flag, but signer flag is a private utility inside `action` package, how can I get access to it?

File: ioctl/newcmd/jwt/jwtsign.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+				return errors.Wrap(err, "failed to sign JWT token")
+			}
+			prvKey.Zero()
+
+			// format output
+			input["iat"] = strconv.FormatInt(now, 10)
+			input["iss"] = "0x" + pubKey.HexString()
+			byteAsJSON, err := json.MarshalIndent(input, "", "  ")
+			if err != nil {
+				log.Panic(err)
```
Comment (CoderZhi) [inline ioctl/newcmd/jwt/jwtsign.go:97]: return err

File: ioctl/newcmd/jwt/jwtsign.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+			prvKey.Zero()
+
+			// format output
+			input["iat"] = strconv.FormatInt(now, 10)
+			input["iss"] = "0x" + pubKey.HexString()
+			byteAsJSON, err := json.MarshalIndent(input, "", "  ")
+			if err != nil {
+				return err
+			}
+			cmd.Printf("JWT token: %s\n\nsigned by:\n{\n  address: %s\n  public key: %s\n}\nwith following claims:\n%s",
```
Comment (huof6829) [inline ioctl/newcmd/jwt/jwtsign.go:97]: it's unreadable.  Split it into some lines ending with "\n" like cmd.Printf("JWT token: %s\n\n"+
"signed by:\n"+
"{\n" +
"    address: %s\n"+

Comment (huof6829) [reply ioctl/newcmd/jwt/jwtsign.go:97]: add  `\n` at the end

**Comment by huof6829** [comment]
pls add test. thx.

**Comment by huof6829** [comment]
> @huof6829 The unittest need to add signer flag, but signer flag is a private utility inside `action` package, how can I get access to it?

hi, I added. Pls see.

File: ioctl/newcmd/jwt/jwt.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	return &cobra.Command{
+		Use:   "jwt",
+		Short: short,
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+
+			cmd.AddCommand(NewJwtSignCmd(client))
+			action.RegisterWriteCommand(client, cmd)
```
Comment (huof6829) [inline ioctl/newcmd/jwt/jwt.go:38]: arg should be `jwtSignCmd`

File: ioctl/newcmd/jwt/jwt.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	short, _ := client.SelectTranslation(_jwtCmdShorts)
+
+	return &cobra.Command{
+		Use:   "jwt",
+		Short: short,
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+
+			cmd.AddCommand(NewJwtSignCmd(client))
```
Comment (huof6829) [inline ioctl/newcmd/jwt/jwt.go:37]: `jwtSignCmd=NewJwtSignCmd(client)`
replace arg of line37 and line 39 with `jwtSignCmd`

**Review by huof6829 — APPROVED** [review]

**Comment by LuckyPigeon** [comment]
@huof6829 Thanks for the improvement!

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**